waku/v2/protocol/waku_store/waku_store.nim

```diff
@@ -254,15 +254,18 @@ proc paginateWithIndex*(list: seq[IndexedWakuMessage], pinfo: PagingInfo): (seq[
   var newCursor: Index # to be returned as part of the new paging info
   case dir
     of PagingDirection.FORWARD: # forward pagination
-      let remainingMessages= uint64(msgList.len) - uint64(foundIndex) - 1
+      # the message that is pointed by the cursor is excluded for the retrieved list, this is because this message has already been retrieved by the querier in its prior request
+      var remainingMessages= uint64(msgList.len) - uint64(foundIndex) - 1 
+      if initQuery:  remainingMessages = remainingMessages + 1
```

Comment:
I wonder if this logic can't be simplified...minor thing though, and if tests are improved now that's great!

Reply:
I agree, was planing to do so in the near future. Here is issue I just opened for this sake https://github.com/status-im/nim-waku/issues/484